frontend/src/metabase-lib/v1/expressions/tokenizer/grammar.lezer

```diff
@@ -0,0 +1,218 @@
+@top Expression { expression }
+
+@external tokens reference from "./tokens" { Reference }
+
+@skip { space }
+
+@precedence {
+  not @left,
+  sum @left,
+  subtract @left,
+  product @left,
+  quotient @left,
+  and @left,
+  or @left,
+  gt @left,
+  gte @left,
+  lt @left,
+  lte @left,
+  eq @left,
+  neq @left
+}
+
+expression {
+  Number
+| String
+| Boolean
+| Reference
+| BinaryExpression
+| ParenExpression
+| CallExpression
+| UnaryExpression
+| SignedExpression
+}
+
+BinaryExpression {
+  expression !sum "+" expression
+| expression !subtract "-" expression
+| expression !product "*" expression
+| expression !quotient "/" expression
+| expression !and And expression
+| expression !or Or expression
+| expression !gt ">" expression
+| expression !gte ">=" expression
+| expression !lt "<" expression
+| expression !lte "<=" expression
+| expression !eq "=" expression
+| expression !neq "!=" expression
+}
+
+UnaryExpression {
+  !not Not expression
+}
+
+SignedExpression {
+  "+" | "-" expression
+}
+
+ParenExpression {
+  "(" expression ")"
+}
+
+CallExpression {
+  Identifier "(" ")"
+| Identifier "(" ArgList ")"
+}
+
+ArgList {
+  Arg ("," Arg)*
+}
+
+Arg {
+  expression
+}
+
+@skip {} {
+  String[isolate] {
+    '"' (stringContentDouble | Escape)* ('"' | "\n")
+  | "'" (stringContentSingle | Escape)* ("'" | "\n")
+  }
+}
+
+
+Not {
+  @specialize<Identifier,
+    "not"
```

Comment:
This is a bit of an ugly wart of the parser grammar, but we cannot specialize tokens in a case-insensitive way.